Simultaneous trajectory execution (#733)
ros-planning/moveit_tutorials: doc/simultaneous_trajectory_execution/simultaneous_trajectory_execution_tutorial.rst

Simultaneous Trajectory Execution
==================================

Introduction
------------

> MoveIt now allows simultaneous execution of trajectories, as long as, each trajectory uses a different set of controllers. For example, in a dual arm environment, each arm can execute a different set of trajectories without needing to wait for the other arm to finish moving or manually synchronizing the motion of both arm into a single trajectory. Optionally, a collision check is performed right before execution of new trajectories to prevent collisions with active trajectories.

Reviewer: ```suggestion
MoveIt allows simultaneous execution of trajectories, as long as each trajectory uses a different set of controllers. For example, in a dual arm environment, each arm can execute a different set of trajectories without needing to wait for the other arm to finish moving or manually synchronizing the motion of both arm into a single trajectory. Optionally, a collision check is performed right before execution of new trajectories to prevent collisions with active trajectories.
```

The following GIF shows simple example of simultaneous execution of trajectories through the **Rviz Motion Planning** plugin.

> .. only:: html

Reviewer: why is that needed?

Author: It is a very quick way to try the simultaneous feature from Rviz.

Reviewer: I was refering to the `.. only:: html` only

Author: Oh, that. I guess it is not necessary, I just copy-paste it from somewhere.

   .. figure:: images/simultaneous-execution-rviz.gif

> This tutorial presents how to use the Simultaneous Trajectory Execution feature through the `Move Group C++ Interface <../move_group_interface/move_group_interface_tutorial.html>`_ but it can be similarly used through the `Move Group Python Interface <../move_group_python_interface/move_group_python_interface_tutorial.html>`_. 

Reviewer: or through MoveIt Cpp?

Author: Okay!

Getting Started
---------------
If you haven't already done so, make sure you've completed the steps in `Getting Started <../getting_started/getting_started.html>`_.

Setup
---------------
The simultaneous trajectory execution feature can be enabled or disabled through the dynamic reconfigure parameter **/move_group/trajectory_execution/enable_simultaneous_execution**.

> Optionally, an extra layer of collision checking, done right before execution of trajectories, can be enabled through the dynamic reconfigure parameter **/move_group/trajectory_execution/enable_collision_checking**.

Reviewer: Both parameters are on by default, here it sounds like they are both off by default

Author: Right, I change the explanation to make it clear that it is on by default, but it is configurable.

Running the code
----------------
Open two shells. In the first shell start RViz and wait for everything to finish loading: ::

  roslaunch moveit_resources_dual_panda_moveit_config demo.launch

In the second shell, run the launch file for this demo: ::

  roslaunch moveit_tutorials simultaneous_trajectory_execution_tutorial.launch

Expected Output
---------------

> For a robotic environment with two robot arms with Franka Panda robots, two different trajectories are planned, one for each robot arm. Then both trajectory can be simultaneously executed.

Reviewer: ```suggestion
In a robotic environment with two Franka Panda robot arms, two different trajectories are planned, one for each robot arm. Then both trajectory are simultaneously executed.
```

Author: Okay!
